dask/core.py

```diff
@@ -304,12 +304,17 @@ def subs(task, key, val):
         elif type_arg is list:
             arg = [subs(x, key, val) for x in arg]
         elif type_arg is type(key):
-            # Can't do a simple equality check, since this may trigger
-            # a FutureWarning from NumPy about array equality
-            # https://github.com/dask/dask/pull/2457
-            if len(arg) == len(key) and all(type(aa) == type(bb) and aa == bb
-                                            for aa, bb in zip(arg, key)):
-                arg = val
+            try:
+                # Can't do a simple equality check, since this may trigger
+                # a FutureWarning from NumPy about array equality
+                # https://github.com/dask/dask/pull/2457
+                if len(arg) == len(key) and all(type(aa) == type(bb) and aa == bb
+                                                for aa, bb in zip(arg, key)):
+                    arg = val
+
+            except (TypeError, AttributeError):
```

Comment:
I recommend adding a small comment here to explain this to future devs: `  # not all keys support len`